dev/PushNotifications/PushNotificationsLongRunningTask/ForegroundSinkManager.h

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include <unordered_map>
+#include <unordered_set>
+#include <algorithm>
+#include <vector>
+
+class ForegroundSinkManager
+{
+public:
+    ForegroundSinkManager() = default;
+
+    void Add(std::wstring processName, IWpnForegroundSink* sink);
+
+    void Remove(std::wstring processName);
+
+    bool InvokeForegroundHandlers(std::wstring processName, byte* payload, ULONG payloadSize);
+
+private:
+    std::unordered_map<std::wstring, Microsoft::WRL::ComPtr<IWpnForegroundSink>> m_foregroundMap = {};
```

Comment:
Please add a comment somewhere to say that "An app can only have one active foreground sink with Long Running Process" and that event handler accounting is done in WinSDK."